### ISSUE TITLE
Internal error: Geometry delta is too big

### ISSUE THREAD

**Issue by jenningsanderson** [mapbox/tippecanoe#833]
I'm curious about the following error: 
`Internal error: Geometry delta is too big: -4188947,2528672683`

This is coming while tiling about 600 million OSM objects (buildings) with the following flags: 
`-Pf -Z14 -z14 -d18 -pf -pk -pg -pD -ps -b0`

Would changing to `z12 -d20` solve this for any reason? 

Thanks!

**Comment by ericfischer** [comment]
This is a surprising error, because it should happen only when a coordinate within a vector tile is further away from the previous coordinate than can fit in a signed 32-bit number, which shouldn't happen with only `-d18`, which should only have coordinates as large as 18 bits.

It must be because `-pD`'s attempt to keep from duplicating features across tiles is putting part of a feature far from its "home" tile. I think in this case you will only find benefit from *lowering* `-d`, not raising it, so that the faraway coordinate can still be a relatively small number.

**Comment by ericfischer** [comment]
I'm not sure what older version it was that you said didn't give this error message, but I only added the check for overflow in version 1.32.9 (Dec 18 2018). Prior to that, it would have overflowed without a warning, corrupting the feature geometry.

**Comment by jenningsanderson** [comment]
Ah, well, that makes sense, then. I rolled back to Version 1.31.0 and the error disappeared... would the corrupted geometries still get tiled? Or were they skipped?

If I had to guess, the features that are causing this to happen include the point [0,0]. (Many OSM features include this point in their histories at some point due to any number of errors); however, this isn't a simple filter to solve it... many OSM features include historical versions in which a node was actually dragged across the city/country/world by accident... I think it's probably these types of features that are causing this error... and I don't have a foolproof way of finding them 🤔 

Would it be possible to output the failing GeoJSON to stderr (and perhaps gracefully skip it?) with something like `--skip-errors` and `--print-errors` for such a case? (Similar to how null geometries are noted, but eventually ignored).

**Comment by ericfischer** [comment]
In the old version, the corrupted geometries would still be tiled, and would just be wrong in the tile.

The error is detected quite late in processing so it would be hard to print the original GeoJSON. It would be possible to print the attributes, though, and to optionally leave out the feature.

**Comment by jenningsanderson** [comment]
Makes sense -- Yes, the ability to optionally leave out the feature instead of stopping completely (and maybe report on the number of failed features at the end and/or print the attributes would be very helpful; I'm not sure if it'd be useful for any other users... but it would save my bizarre use cases.
